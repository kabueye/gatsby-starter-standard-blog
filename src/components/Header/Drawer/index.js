// @flow strict
export { default } from './Drawer';
