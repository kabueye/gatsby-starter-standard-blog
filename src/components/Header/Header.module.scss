@import '../../assets/scss/variables';
@import '../../assets/scss/mixins';

.header {
    &__title {
        font-size: $typographic-base-font-size * 2;
        color: $color-white;
        margin-right: auto;
        font-weight: 600;
    }
}
