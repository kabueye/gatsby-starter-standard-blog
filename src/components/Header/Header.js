// @flow strict
import React from 'react';
import { AppBar, Toolbar } from '@material-ui/core';
import { Link } from 'gatsby';

import { useSiteMetadata } from '../../hooks';
import styles from './Header.module.scss';
import Menu from './Menu';
import Drawer from './Drawer';

const Header = () => {
  const { title, menu } = useSiteMetadata();

  return (
    <AppBar position="static">
      <Toolbar>
        <Link to="/" className={styles.header__title}>
          {title}
        </Link>
        <Menu menu={menu} />
        <Drawer menu={menu} />
      </Toolbar>
    </AppBar>
  );
};

export default Header;
