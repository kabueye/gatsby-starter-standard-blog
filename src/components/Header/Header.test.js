// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import { useStaticQuery, StaticQuery } from 'gatsby';
import siteMetadata from '../../../jest/__fixtures__/site-metadata';
import type { RenderCallback } from '../../types';
import Header from './Header';

describe('Header', () => {
  beforeEach(() => {
    StaticQuery.mockImplementationOnce(
      ({ render }: RenderCallback) => render(siteMetadata),
      useStaticQuery.mockReturnValue(siteMetadata),
    );
  });

  const props = {
    children: 'test',
    title: 'test',
  };

  it('renders correctly', () => {
    const tree = renderer.create(<Header {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
