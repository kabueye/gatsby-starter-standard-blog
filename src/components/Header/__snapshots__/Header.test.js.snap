// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
  >
    <a
      className="header__title"
      href="/"
    >
      Test title
    </a>
    <nav
      className="menu"
    >
      <a
        className="menu__link"
        href="/test/1/"
      >
        Test label 1
      </a>
      <a
        className="menu__link"
        href="/test/2/"
      >
        Test label 2
      </a>
      <a
        className="menu__link"
        href="/test/3/"
      >
        Test label 3
      </a>
    </nav>
    <nav
      className="drawer"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root drawer__icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
      </button>
    </nav>
  </div>
</header>
`;
